services/src/main/java/org/keycloak/services/managers/ClientManager.java

```diff
@@ -355,4 +357,7 @@ private boolean showClientCredentialsAdapterConfig(ClientModel client) {
         return authenticator.getAdapterConfiguration(client);
     }
 
+    private boolean isInternalClient(Stream<RealmModel> stream, String clientId) {
+        return stream.anyMatch(realm -> clientId.equals(realm.getName() + "-realm"));
+    }
```

Comment:
There is no need to go through all existing realms here, with large number of realms it could become costly operation. Can it be used `session.realms().getRealmByName()` instead?

Reply:
Fixed. The realm name is now parsed from the client (if a "-realm" suffix is present) and thus only a single query is made.